src/lua/utils.c

```diff
@@ -735,6 +735,22 @@ luaT_newthread(struct lua_State *L)
 	return L1;
 }
 
+/** A getenv wrapper checking for size of the corresponding env variable. */
+static int
+luaT_getenv_safe(struct lua_State *L)
+{
+	if (lua_gettop(L) != 2 || !lua_isstring(L, 1) || !lua_isnumber(L, 2))
+		return luaL_error(L, "usage: getenv_safe(name, length)");
+	const char *name = lua_tostring(L, 1);
+	int len = lua_tonumber(L, 2);
+	const char *env = getenv_safe(name, len, NULL);
```

Comment:
It means 'limit the string to `len-1`' and seems a bit counter-intuitive for Lua. Shouldn't we limit the string length to the argument value (IOW, call `getenv_safe(name, len + 1, NULL)`)?

Reply:
It is a bit counter-intuitive, indeed. But my thoughts were that `os.getenv_safe` will be mostly used by lua code which intends to pass the env values to C later on. So it has to know about "size" vs "len" differences.

And since it's not clear what users will pass there, it's better to be on the safe side.

Reply:
I would ask to clarify it in the documentation request and, maybe, a comment.

Reply:
Sure:
```diff
diff --git a/src/lua/env.lua b/src/lua/env.lua
index 9a505c6b3..288e91ac2 100644
--- a/src/lua/env.lua
+++ b/src/lua/env.lua
@@ -40,6 +40,13 @@ os.setenv = function(key, value)
     end
 end
 
+--
+-- Return the environment variable pointed to by key, as long as its length is
+-- strictly less than size.
+-- Note, an environment variable with length == size will not be returned,
+-- because size accounts an additional byte for a C string null terminator
+-- (the user would probably care about exact env value size only if he intends
+-- to pass the value to C).
+--
 os.getenv_safe = function(key, size)
     local ret = ffi.C.getenv_safe(key, size, nil)
     if ret ~= nil then
```

I started rewriting the doc request but then decided that there's no need to go into details there. Let's simply state that the returned value will have length strictly less than the passed value.